gitoxide-core/src/organize.rs

```diff
@@ -72,21 +75,13 @@ where
 }
 
 fn find_origin_remote(repo: &Path) -> anyhow::Result<Option<git_url::Url>> {
-    let out = std::process::Command::new("git")
-        .args(&["remote", "--verbose"])
-        .current_dir(repo)
-        .output()?;
-    if out.status.success() {
-        Ok(parse::remotes_from_git_remote_verbose(&out.stdout)?
-            .into_iter()
-            .find_map(|(origin, url)| if origin == "origin" { Some(url) } else { None }))
-    } else {
-        anyhow::bail!(
-            "git invocation failed with code {:?}: {}",
-            out.status.code(),
-            out.stderr.as_bstr()
-        )
-    }
+    let mut config_bytes = vec![];
+    let config = {
+        let mut file = std::fs::File::open(repo.join("./.git/config"))?;
+        file.read_to_end(&mut config_bytes)?;
+        GitConfig::try_from(&config_bytes).map_err(|e| e.to_owned())?
```

Comment:
Idk if we want the error type to just always own the bytes or not. If we do then we don't need to have the `map_err` call... On the other hand if we do we're guaranteeing a performance difference between a normal case or on an error, since we might alloc a potentially very large string.

Reply:
That looks good to me, but if the alternative would be more readable it could be preferable.